patches/server/0887-Add-option-to-hide-entities-by-default.-This-option-.patch

```diff
@@ -0,0 +1,118 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: CrystallDEV <m.hasselder@gmx.de>
+Date: Sun, 27 Mar 2022 15:05:06 +0200
+Subject: [PATCH] Add option to hide entities by default. This option is
+ currently very strict and overrides all other changes to visibility and
+ interaction. Fixes #7395
+
+
+diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
+index 3b1a34b34979ab436ccd33f0a85bfae537cbecb4..d558883804d450d39750e5b2f4d2cf4cd021ebfb 100644
+--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
++++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
+@@ -1524,6 +1524,11 @@ public class ServerLevel extends Level implements WorldGenLevel {
+ 
+     @Override
+     public void playSound(@Nullable Player except, Entity entity, SoundEvent sound, SoundSource category, float volume, float pitch) {
++        // Paper start - dont play sounds of entities that are hidden by default
++        if(entity.isHiddenEntityByDefault()){
++            return;
++        }
```

Comment:
I would instead return if the entity is hidden at all (CraftPlayer#canSee)

Reply:
This would require to have the player that is listening to the sound, which would require me to modify the broadcast method and therefore have a completely different implementation

Reply:
Ahh, my bad I misunderstood the method parameters. Disregard my comment.